### PR TITLE
Disable deprecated implementation warnings for Tests

### PR DESCRIPTION
**Pull request by sergiocampama**

### CODE REVIEW

**Comment by sergiocampama** [comment]
@thomasvl  new implicit warning passed on the new beta... disable for the tests about the unittest_deprecated.proto messages.

**Comment by thomasvl** [comment]
So #2627 went away?  i.e. - we should close that?

What exactly warns in on the deprecated side of things?  We want the generated .m files to now warn for things they are defining (so the generated code always compiles cleanly).  So I'm not sure suppressing is what we want if we're using this to track that the code works correctly.

File: src/google/protobuf/compiler/objectivec/objectivec_file.cc [hunk 1]
```diff
@@ -217,14 +217,16 @@ void FileGenerator::GenerateHeader(io::Printer *printer) {
...
+  //  deprecated-declarations and deprecated-implementations suppression is 
+  //    only needed when any symbol in this proto file has something 
+  //    deprecated or if it references something from another file that is
+  //    deprecated.
   printer->Print(
       "// @@protoc_insertion_point(imports)\n"
       "\n"
       "#pragma clang diagnostic push\n"
       "#pragma clang diagnostic ignored \"-Wdeprecated-declarations\"\n"
+      "#pragma clang diagnostic ignored \"-Wdeprecated-implementations\"\n"
```
Comment (thomasvl) [inline src/google/protobuf/compiler/objectivec/objectivec_file.cc]: Why'd you do this in the header, isn't it only needed in the .m file?

**Comment by sergiocampama** [comment]
Can't reproduce any more

**Pull request closed without merging**